### PR TITLE
Initial support for Nitrogen8MM DWE

### PR DESCRIPTION
**Pull request by acostach**
Add initial support for N8MM DWE device type.

### CODE REVIEW

File: layers/meta-balena-fsl-arm/conf/samples/local.conf.sample [hunk 1]
```diff
@@ -2,13 +2,16 @@
...
+#MACHINE ?= "nitrogen8mm"
+#MACHINE ?= "nitrogen8mm-dwe"
 
 # More info meta-resin/README.md
 #TARGET_REPOSITORY ?= ""
 #TARGET_TAG ?= ""
 
 BALENA_STORAGE_nitrogen6xq2g = "overlay2"
 BALENA_STORAGE_nitrogen8mm = "overlay2"
+BALENA_STORAGE_nitrogen8mm-dwe = "overlay2"
```
Comment (floion) [inline layers/meta-balena-fsl-arm/conf/samples/local.conf.sample]: The git commit log is misleading a little as it implies we switched to overlay2 for other machines that were using something else before this change

File: layers/meta-balena-fsl-arm/conf/machine/nitrogen8mm-dwe.conf [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+#@SOC: i.MX8 MQ
+#@DESCRIPTION: Machine configuration for Boundary Devices Nitrogen8MM DWE
+#@MAINTAINER: Chris Dimich <chris.dimich@boundarydevices.com>
+
+MACHINEOVERRIDES = "nitrogen8mm:${MACHINE}"
+
+include conf/machine/nitrogen8mm.conf
+
+# Kernel configuration
+KERNEL_DEVICETREE += " \
```
Comment (floion) [inline layers/meta-balena-fsl-arm/conf/machine/nitrogen8mm-dwe.conf]: Do we need the other dtbs too for this machine?

File: layers/meta-balena-fsl-arm/recipes-bsp/u-boot/u-boot-boundary_%.bbappend [hunk 3]
```diff
@@ -3,8 +3,8 @@ inherit resin-u-boot
 
 FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
 
-# Use latest revision
-SRCREV="85f75fa9e9db29f45ef5fcee0508973f4ad82698"
+# Override bsp layer autorev and use latest revision
+SRCREV="b5b56bb07fc8f8ec541e58dc56aa48bed0ea3724"
```
Comment (floion) [inline layers/meta-balena-fsl-arm/recipes-bsp/u-boot/u-boot-boundary_%.bbappend]: How was the BSP using autorev? I see they have it fixed also

File: layers/meta-balena-fsl-arm/recipes-bsp/imx-mkimage/imx-mkimage%.bbappend [hunk 4]
```diff
@@ -1,3 +1,3 @@
 # Use latest and known to work revision. Otherwise if
 # booting fails at some point we won't know what caused it
-SRCREV = "2bb49a37e1d964eacebb09aa0073048a2bfd26e8"
+SRCREV = "dc22e26d07f249325604f7dfd42324a9a75e7cc0"
```
Comment (floion) [inline layers/meta-balena-fsl-arm/recipes-bsp/imx-mkimage/imx-mkimage%.bbappend]: Same question about autorev as for the bootloader

File: layers/meta-balena-fsl-arm/recipes-bsp/u-boot/u-boot-boundary_%.bbappend (hunk 3 above)
Comment (acostach) [reply layers/meta-balena-fsl-arm/recipes-bsp/u-boot/u-boot-boundary_%.bbappend]: @floion we're using master branch: https://github.com/boundarydevices/meta-boundary/commit/1e76e65c2c22912fa99fc9f535b7e4c75b064112

File: layers/meta-balena-fsl-arm/recipes-bsp/imx-mkimage/imx-mkimage%.bbappend (hunk 4 above)
Comment (acostach) [reply layers/meta-balena-fsl-arm/recipes-bsp/imx-mkimage/imx-mkimage%.bbappend]: We're using the master branch, where all these are autorev.

File: layers/meta-balena-fsl-arm/conf/samples/local.conf.sample (hunk 1 above)
Comment (acostach) [reply layers/meta-balena-fsl-arm/conf/samples/local.conf.sample]: Re-worded, thanks

File: layers/meta-balena-fsl-arm/conf/machine/nitrogen8mm-dwe.conf (hunk 2 above)
Comment (acostach) [reply layers/meta-balena-fsl-arm/conf/machine/nitrogen8mm-dwe.conf]: Correct, I removed them from here and from the boot partition too.

**Pull request merged**